src/actions.js

```diff
@@ -149,11 +161,47 @@ function optSetIncDecStep(label = 'Value', def, min, max, step = 1) {
 			min: 1,
 			max: max - min,
 			required: true,
-			isVisible: (options) => options.op !== 's',
+			isVisible: (options) => options.op !== 's' && !options.useVar,
+		},
+		{
+			id: 'stepVar',
+			type: 'textinput',
+			label: 'Step Size',
+			default: `${step}`,
+			regex: Regex.SOMETHING,
+			required: true,
+			useVariables: true,
+			tooltip: `Variable should return a number between ${1} and ${max - min}`,
+			isVisible: (options) => options.op !== 's' && options.useVar,
+		},
+		{
+			id: 'useVar',
+			type: 'checkbox',
+			label: 'Use Variable',
+			default: false,
 		},
 	]
 }
 
+async function parseSetIncDecVariables(action, self, min, max, stepSize = 1) {
+	let set = action.options.useVar ? Number(await self.parseVariablesInString(action.options.setVar)) : action.options.set
+	let step = action.options.useVar ? Number(await self.parseVariablesInString(action.options.stepVar)) : action.options.step
+	if (action.options.op === ACTION_SET && isNaN(set)) {
+		self.log('debug', `Invalid set variable ${set} from ${action.options.setVar}. Must be between ${min} and ${max}`)
+		return undefined
+	}
+	if (action.options.op !== ACTION_SET && isNaN(step)) {
+		self.log('debug', `Invalid step variable ${step} from ${action.options.stepVar}. Must be between 1 and ${max - min}`)
+		return undefined
+	}
+	// if variables exceeed limits conform to limit rather than rejecting.
+	set = set > max ? max : set < min ? min : set
+	step = step > max - min ? max - min : step < 1 ? 1 : step
+	action.options.set = set - (set % stepSize)
```

Comment:
Is there any situation where this would be invalid? 
If `min % stepSize !== 0` it may not work as desired.